nibabel/tests/test_scripts.py

```diff
@@ -54,15 +54,62 @@ def script_test(func):
 
 
 @script_test
-def test_nib_ls():
+def _test_nib_ls_example4d(opts=[], hdrs_str=""):
     # test nib-ls script
     fname = pjoin(DATA_PATH, 'example4d.nii.gz')
     expected_re = (" (int16|[<>]i2) \[128,  96,  24,   2\] "
-                   "2.00x2.00x2.20x2000.00  #exts: 2 sform$")
-    cmd = ['nib-ls', fname]
+                   "2.00x2.00x2.20x2000.00  #exts: 2%s sform$"
+                   % hdrs_str)
+    cmd = ['nib-ls'] + opts + [fname]
     code, stdout, stderr = run_command(cmd)
     assert_equal(fname, stdout[:len(fname)])
-    assert_not_equal(re.match(expected_re, stdout[len(fname):]), None)
+    assert_re_in(expected_re, stdout[len(fname):])
+
+@script_test
+def test_nib_ls():
+    yield _test_nib_ls_example4d
+    yield _test_nib_ls_example4d, ['-H', 'dim,bitpix'], " \[  4 128  96  24   2   1   1   1\] 16"
+
+@script_test
+def test_nib_ls_multiple():
+    # verify that correctly lists/formats for multiple files
+    fnames = [
+        pjoin(DATA_PATH, f)
+        for f in ('example4d.nii.gz', 'example_nifti2.nii.gz', 'small.mnc', 'nifti2.hdr')
+    ]
+    code, stdout, stderr = run_command(['nib-ls'] + fnames)
+    stdout_lines = stdout.split('\n')
+    assert_equal(len(stdout_lines), 4)
+    # they should be indented correctly.  Since all files are int type -
+    ln = max(len(f) for f in fnames)
+    assert_equal([l[ln:ln+2] for l in stdout_lines], [' i']*4,
```

Comment:
actually trimming some would only bring disturbance into the field for no benefit (e.g. matching those output lines with stats in them), so I will let those stay as is